presto-main/src/test/java/com/facebook/presto/operator/aggregation/TestApproximatePercentileAggregation.java

```diff
@@ -194,5 +290,35 @@ public void testDoublePartialStep()
                 createDoublesBlock(1.0, null, 2.0, null, 2.0, null, 2.0, null, 3.0, null, 3.0, null, 3.0, 4.0, 5.0, 6.0, 7.0),
                 createLongsBlock(1L, 1L, 2L, 1L, 2L, 1L, 2L, 1L, 2L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L),
                 createRLEBlock(0.5, 17));
+
+        // weighted + array of approx_percentile
+        assertAggregation(
+                DOUBLE_APPROXIMATE_PERCENTILE_ARRAY_WEIGHTED_AGGREGATION,
+                1.0,
+                ImmutableList.of(1.0, 2.0),
+                createDoublesBlock(1.0, 2.0, 3.0),
+                createLongsBlock(4L, 2L, 1L),
+                createRLEBlock(new Double[] {0.5, 0.8}, 3));
+    }
+
+    private static RunLengthEncodedBlock createRLEBlock(double percentile, int positionCount)
+    {
+        BlockBuilder blockBuilder = DOUBLE.createBlockBuilder(new BlockBuilderStatus(), 1);
+        DOUBLE.writeDouble(blockBuilder, percentile);
+        return new RunLengthEncodedBlock(blockBuilder.build(), positionCount);
+    }
+
+    private static RunLengthEncodedBlock createRLEBlock(Double[] percentiles, int positionCount)
```

Comment:
Generally, we use `Iterable<Double>` instead of `Double[]` in such cases. And we use `ImmutableList.of` at call sites.